### ISSUE TITLE
[HOLD for payment 2024-04-09] [$500] Expense - Page scrolls back down when dismissing add receipt RHP in transaction thread

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#38935]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.56-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Create a manual request without receipt
3. Go to request details page
4. Scroll to the top of the page
5. Click on add receipt thumbnail
6. Click outside the RHP to dismiss it

## Expected Result:
The page will not scroll down when closing the RHP (production behavior)

## Actual Result:
The page scrolls down when closing the RHP

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/8806e554-6a84-4c05-84db-b1feb30c547a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e4b313e99e0a0eca</li>
        <li>Upwork Job ID: 1772334661375188992</li>
        <li>Last Price Increase: 2024-03-25</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mananjadhav</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by lanitochka17** [comment]
@flodnv  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by Habtamu-Asefa** [comment]
I couldn't reproduce this on dev but it is still there on staging.

**Comment by godofoutcasts94** [comment]
If it is caused by this [PR](https://github.com/Expensify/App/pull/38027) then is it not a regression?

**Comment by flodnv** [comment]
+1, are we saying the bug is reproducible on production?

**Comment by flodnv** [comment]
I can confirm this bug is reproducible only in staging and not on production.

**Comment by yuwenmemon** [comment]
Taking over for @flodnv who needs to run

**Comment by yuwenmemon** [comment]
The proposal by @abzokhattab is not the solution, we still call `OpenReport` on production just fine when dismissing the modal: 
![Screenshot 2024-03-25 at 3 22 22 PM](https://github.com/Expensify/App/assets/4741899/69bc5ad2-b254-4ed1-8900-d9c453bfdd32)

**Comment by yuwenmemon** [comment]
@abzokhattab Fair point, but I don't think that's a concern for this issue. Feel free to bring up that concern in #expensify-open-source 🙂

**Comment by yuwenmemon** [comment]
Still looking for reasons why the scrolling is happening and possible solutions, but this is a minor issue so I won't let this block the deploy.

**Comment by perunt** [comment]
Okay, so just to recap, the issue occurs when we call `OpenReport` upon clicking outside the RHP. [The PR that introduced this behavior](https://github.com/Expensify/App/pull/38027/files#diff-68b13bc8344820144538408415f307efee198ccc63977ba318c4234102ec3917) seems to be the source. It appears the `prevIsFocused` value acts differently when the screen is wide or small, resulting in `OpenReport` being called when there is no need for it.
Here `prevIsFocused` changed twice when i close the window:
<img width="1111" alt="image" src="https://github.com/Expensify/App/assets/48593211/5f56e850-7119-4531-97dd-b22b9b6a3d11">

Firstly, it's important to note that [this fix](https://github.com/Expensify/App/pull/38027/files#diff-68b13bc8344820144538408415f307efee198ccc63977ba318c4234102ec3917) is only valid for mobile devices. The bug was originally a mobile-only issue based on my testing. The behavior of `prevIsFocused` differs between web and mobile platforms, which means it's not a suitable basis for this fix.

I tried using `const join = Session.checkIfActionIsAllowed(() => {...`, and it successfully receives all necessary props. This leads me to believe that we can address the issue without resorting to calling `OpenReport`.

Given the numerous blockers we're facing, I need to jump on them :). Therefore, as a temporary solution, I suggest adding `isSmallScreenWidth` to [this line in ReportScreen.tsx](https://github.com/Expensify/App/blob/d8361eee3486ccb5472060f2af7655e96461471e/src/pages/home/ReportScreen.tsx#L432). This adjustment would prevent `OpenReport` from being called and eliminate unnecessary scrolling on wide screens. For small screens, scrolling might still occur if there are no messages following this report action, but this is a rare case. Even if it happens, the scrolling is not significant—typically, it's up to half the screen height.

So, TL;DR: We've gotta avoid calling `OpenReport`, I believe, or better yet, cut it out entirely. For now, I’m pitching a one-liner as a temporary fix. It doesn’t introduce any new 'bad' code, just extends an existing hack.
cc @yuwenmemon

**Comment by abzokhattab** [comment]
thanks @perunt, that's a very good observation, i have updated my proposal to check for the `isSmallScreenWidth ` in the condition.and its seems to fix the issue for me as well.

**Comment by perunt** [comment]
It doesn't fix the origin of the issue, which I believe is [this PR](https://github.com/Expensify/App/pull/38027/files#diff-68b13bc8344820144538408415f307efee198ccc63977ba318c4234102ec3917)
@roryabraham if you're good with it, we can fix with [PR](https://github.com/Expensify/App/pull/38999/files)

**Comment by yuwenmemon** [comment]
@perunt Sounds good to me - can we prep the PR for review?

**Comment by abzokhattab** [comment]
@yuwenmemon @perunt let me know what you think about the alternative proposal [here](https://github.com/Expensify/App/issues/38935#issuecomment-2018815973), it fixes [the join issue](https://github.com/Expensify/App/issues/37612) in a different way ... let me know also if you need the implementation details.

**Comment by kadiealexander** [comment]
@mananjadhav please complete the checklist!

**Comment by mananjadhav** [comment]
Thanks for the bump @kadiealexander. We already discussed the offending [PR](https://github.com/Expensify/App/pull/38027/files#diff-68b13bc8344820144538408415f307efee198ccc63977ba318c4234102ec3917).

I don't think we need a regression test for this one.

**Comment by JmillsExpensify** [comment]
$500 approved for @mananjadhav based on checklist above.
